core/lib/l1_contract_interface/src/i_executor/structures/commit_batch_info.rs

```diff
@@ -101,15 +104,81 @@ impl<'a> Tokenizable for CommitBatchInfo<'a> {
                 ),
                 // `systemLogs`
                 Token::Bytes(serialize_commitments(&self.0.header.system_logs)),
-                // `totalL2ToL1Pubdata`
-                Token::Bytes(
-                    self.0
-                        .header
-                        .pubdata_input
-                        .clone()
-                        .unwrap_or(self.0.construct_pubdata()),
+            ]
+        }
+    }
+}
+
+impl<'a> Tokenizable for CommitBatchInfo<'a> {
+    fn from_token(_token: Token) -> Result<Self, Web3ContractError>
+    where
+        Self: Sized,
+    {
+        // Currently there is no need to decode this struct.
+        // We still want to implement `Tokenizable` trait for it, so that *once* it's needed
+        // the implementation is provided here and not in some other inconsistent way.
+        Err(Web3ContractError::Api(Web3ApiError::Decoder(
+            "Not implemented".to_string(),
+        )))
+    }
+
+    fn into_token(self) -> Token {
+        let mut tokens = self.base_tokens();
+
+        if !self.0.header.protocol_version.unwrap().is_pre_boojum() {
+            match self.1 {
+                None => tokens.push(
+                    // `totalL2ToL1Pubdata` without pubdata source byte
+                    Token::Bytes(
+                        self.0
+                            .header
+                            .pubdata_input
+                            .clone()
+                            .unwrap_or(self.0.construct_pubdata()),
+                    ),
                 ),
-            ])
+                Some(pubdata_da) => match pubdata_da {
+                    PubdataDA::Calldata => {
+                        let pubdata = self
+                            .0
+                            .header
+                            .pubdata_input
+                            .clone()
+                            .unwrap_or(self.0.construct_pubdata());
+
+                        let blob_commitment =
+                            KzgInfo::new(self.2.as_ref().unwrap(), &pubdata).to_blob_commitment();
+
+                        let result = std::iter::once(PUBDATA_SOURCE_CALLDATA)
+                            .chain(pubdata)
+                            .chain(blob_commitment)
+                            .collect();
+
+                        tokens.push(Token::Bytes(result));
+                    }
+                    PubdataDA::Blobs => {
+                        // `pubdataCommitments` with pubdata source byte
+                        let pubdata = self
```

Comment:
this pubdata fetching is same for both branches -  maybe move it to line 140 before the match